includes/wc-core-functions.php

```diff
@@ -1032,7 +1032,15 @@ function wc_print_js() {
  */
 function wc_setcookie( $name, $value, $expire = 0, $secure = false, $httponly = false ) {
 	if ( ! headers_sent() ) {
-		setcookie( $name, $value, $expire, COOKIEPATH ? COOKIEPATH : '/', COOKIE_DOMAIN, $secure, apply_filters( 'woocommerce_cookie_httponly', $httponly, $name, $value, $expire, $secure ) );
+		setcookie(
+			$name,
+			$value,
+			$expire,
+			COOKIEPATH ? COOKIEPATH : '/',
+			COOKIE_DOMAIN,
+			apply_filters( 'woocommerce_cookie_secure', $secure, $name, $value, $expire, $httponly ),
+			apply_filters( 'woocommerce_cookie_httponly', $httponly, $name, $value, $expire, $secure )
+		);
```

Comment:
```suggestion
		$options = array(
			'expires'  => $expire,
			'path'     => COOKIEPATH ? COOKIEPATH : '/',
			'domain'   => COOKIE_DOMAIN,
			'secure'   => $secure,
			'httponly' => apply_filters( 'woocommerce_cookie_httponly', $httponly, $name, $value, $expire, $secure ),
		);
		$options = apply_filters( 'woocommerce_cookie_option', $options );
		setcookie( $name, $value, $options['expires'], $options['path'], $options['domain'], $options['secure'], $options['httponly'] );
```

In PHP 7.3, this function definition also allows sending all the params as an object. Instead of adding filters for each param, perhaps we can make the function compatible like in the suggestion with PHP 7.3 style declaration. WDYT?

Reply:
I think the added flexibility would be a great benefit and make this more usable for different requirements. As far as I can judge this would be almost identical to PR https://github.com/woocommerce/woocommerce/pull/28105 sans the check for the PHP version. Would your proposed change also work with PHP < 7.3?